### ISSUE TITLE
Can't resolve relative imports

### ISSUE THREAD

**Issue by spindensity** [microsoft/pyright#124]
Pyright can't resolve relative imports as below correctly.
 
They look like the same problems as #58 and #59, maybe there're some regressions. 

```python
from . import some_class
from .some_module import some_func
```

Pyright 1.0.27 with VS Code 1.33.1 and Python 3.7 virtual environment.

**Comment by erictraut** [comment]
Can you provide more repro steps? What other files or subdirectories are present in the same directory as this file?

**Comment by spindensity** [comment]
After some digging, they are not the problems as #58 and #59. These problems only happen when there's `pyrightconfig.json` in the project root folder. Without that config file, everything is OK.

Minimal test case:

Project structure:
![01](https://user-images.githubusercontent.com/33576822/57588948-16638d80-754f-11e9-82a0-3acb4241eeac.png)

main.py:
```python
if __name__ == "__main__":
    pass
```

`some_package/__init__.py`:
```python
class some_class:
    pass
```

`some_package/module1.py`:
```python
def some_func():
    pass
```

`some_package/module2.py`:
```python
from . import some_class
from .module1 import some_func


some_obj = some_class()
some_func()
```

`pyrightconfig.json`:
```json
{
    "venvPath": "C:/Dev/BasicTools/python/miniconda3/envs",
    "venv": "datascience"
}
```

The virtual environment:
![02](https://user-images.githubusercontent.com/33576822/57588997-d8b33480-754f-11e9-863f-47d98a7fde0d.png)

Result:
![03](https://user-images.githubusercontent.com/33576822/57589009-24fe7480-7550-11e9-827e-082d235d07c7.png)
![04](https://user-images.githubusercontent.com/33576822/57589012-26c83800-7550-11e9-8245-a63e38ad24ba.png)

**Comment by erictraut** [comment]
Thanks for the details, but I'm still not able to repro what you're seeing. There must be something else I'm missing. Could you open the Output window and look at the Pyright output? There should be some additional information provided there. Look for something like "Attempting to resolve relative import". I'm interested in the lines that follow. Thanks!

**Comment by spindensity** [comment]
OK.

Pyright output:

```
Pyright language server starting
Received updated settings
Loading configuration file at c:\Dev\Projects\github-bug-reports\pyright\bug_id_124\pyrightconfig.json
The venvPath has been specified in both the config file and the VS Code settings. The value in the config file (C:\Dev\BasicTools\python\miniconda3\envs) will take precedence
Searching for source files
No source files found.
Could not import '.' in file '/c%3A/Dev/Projects/github-bug-reports/pyright/bug_id_124/some_package/module2.py'
  Attempting to resolve relative import
  Attempting to resolve using root path ''
  Partially resolved import with directory ''
Could not import '.module1' in file '/c%3A/Dev/Projects/github-bug-reports/pyright/bug_id_124/some_package/module2.py'
  Attempting to resolve relative import
  Attempting to resolve using root path ''
  Could not find directory 'module1'
  Did not find file 'module1.pyi' or 'module1.py'
```

**Comment by erictraut** [comment]
OK, thanks. It looks like it's a Windows-specific path issue. That explains why I wasn't seeing it, since my main development machine is a Mac. I'll switch over to a Windows machine and try to repro it there.

**Comment by spindensity** [comment]
Thank you.

After checking the output of pyright, it's obvious that these problems are still here, they just can't be seen from the editor panel but the output message is the same.

Pyright output message without the config file:
![05](https://user-images.githubusercontent.com/33576822/57589991-094c9b80-755b-11e9-9393-46862e8575c4.png)

**Comment by erictraut** [comment]
Found the problem. As I suspected, it had to do with processing Windows-specific file paths with the old DOS-style drive letter at the start. I've switched to a standard library for doing the parsing in a cross-platform way. This should be fixed in the next version of pyright.

Thanks again for reporting the bug!

**Comment by erictraut** [comment]
This should now be fixed in version 1.0.28, which I just published. Thanks for reporting the issue!
